### DISCUSSION TITLE
How to hydrate api response headers for pagination ?

### DISCUSSION THREAD

**Discussion by michael-picard** [nuxt/nuxt#17432] [Questions] [answered]
Hello there, 

I'm fetching products from an api and the pagination data is included in the response headers. The issue is that with the code below the headers are not passed to the browser and the `paginationHeaders` value doesn't get hydrated like `products` value does.

Would a plugin be the correct way to handle this ? I was thinking about hooking into `nuxtApp.payload` to add a headers property.

```
const wooFetch = useWooFetchInstance()
const paginationHeaders = ref<Headers>(null)

const {
  pending: pendingProducts,
  data: products,
} = useLazyAsyncData(`product-${taxo.value.route}-${route}-products`, () => {
  if (process.server && taxonomy.value.id > 0) {
    return wooFetch.raw(`/wc-g2r/v2/products?${taxo.value.param}=${taxonomy.value.id}&per_page=8&status=publish`)
        .then(response => {
          paginationHeaders.value = response.headers
          return response
        })
        .then(response => response._data)
  }
  return []
})
```

**Comment by Flanker72** [comment, marked as answer]
You may try to use `useState` instead of `ref` for `paginationHeaders` - it is SSR-friendly replacement for refs, which should be hydrated.

**Comment by michael-picard** [comment]
Thanks @Flanker72, it does work with `useState`. I've also tried `useHydration` as from what I understood from source code it is meant for this use case, but could never get the `app:created` hook to trigger...

**Reply by Flanker72** [reply]
`app:created` hook works when Vue app is created during SSR. This hook triggers ONLY when the page is requested, and it's triggered on ANY render. You'll never reach this hook just after server start.

**Comment by michael-picard** [comment]
Actually `Headers` interface is not serializable and it has to be converted into a regular object for `useState` to work apparently. I use `Object.fromEntries(response.headers.entries())`.
